--- src/Oro/Bundle/DistributionBundle/OroDistributionBundle.php	2019-08-08 19:23:39.290000000 +0300
+++ src/Oro/Bundle/DistributionBundle/OroDistributionBundle.php	2020-11-23 13:27:06.967945659 +0200
@@ -2,6 +2,8 @@
 
 namespace Oro\Bundle\DistributionBundle;
 
+use Monolog\ErrorHandler;
+use Psr\Log\LogLevel;
 use Oro\Bundle\DistributionBundle\DependencyInjection\Compiler\CacheConfigurationPass;
 use Oro\Bundle\DistributionBundle\DependencyInjection\Compiler\HiddenRoutesPass;
 use Oro\Bundle\DistributionBundle\DependencyInjection\Compiler\RoutingOptionsResolverPass;
@@ -24,4 +26,11 @@
         $container->addCompilerPass(new RoutingOptionsResolverPass());
         $container->addCompilerPass(new HiddenRoutesPass());
     }
+
+    public function boot()
+    {
+        if (PHP_SAPI === 'cli') {
+            ErrorHandler::register($this->container->get('logger'), false, false, LogLevel::CRITICAL);
+        }
+    }
 }
