--- src/Symfony/Component/Debug/ErrorHandler.php	2018-04-06 17:52:23.000000000 +0300
+++ src/Symfony/Component/Debug/ErrorHandler.php	2019-07-16 19:31:03.000000000 +0300
@@ -123,7 +123,7 @@
     public static function register($handler = null, $replace = true)
     {
         if (null === self::$reservedMemory) {
-            self::$reservedMemory = str_repeat('x', 10240);
+            self::$reservedMemory = str_repeat('x', 32 * 1024 * 1024);
             register_shutdown_function(__CLASS__.'::handleFatalError');
         }
 
